src/cmstats.c

```diff
@@ -49,6 +50,26 @@ s_duplicator (const void *self)
     return (void*) fty_proto_dup ((fty_proto_t*) self);
 }
 
+static double
+safe_atof (const char *str)
+{
+    char *endptr = (char *) zmalloc (sizeof (char));
+    errno = 0;
+    double ret = strtod (str, &endptr);
+    // str does not contain number
+    if (ret == 0 && endptr == str) {
+        log_error ("strtod received non-number argument %s", str);
+        //ret = "NAN";
+    }
+    // overflow or underflow
+    if (ret == 0 && errno == ERANGE) {
+        log_error ("strtod experienced over/underflow when converting %s", str);
+        //ret = "NAN";
+    }
+
+    free (endptr);
+    return ret;
```

Comment:
Should we return the zero number, like before? Is the added benefit here at the moment the logging in case of error, so we can at least have some debug log when issues happen? Maybe also log the string at `endptr` if it is not NULL and `endptr[0] != '\0'` to see if some wrong text got into messages?